omegaconf/dictconfig.py

```diff
@@ -146,41 +146,15 @@ def _validate_get(self, key: Any, value: Any = None) -> None:
                     key=key, value=value, cause=ConfigAttributeError(msg)
                 )
 
-    def _validate_merge(self, key: Any, value: Any) -> None:
-        self._validate_set_merge_impl(key, value, is_assign=False)
-
     def _validate_set(self, key: Any, value: Any) -> None:
-        self._validate_set_merge_impl(key, value, is_assign=True)
-
-    def _validate_set_merge_impl(self, key: Any, value: Any, is_assign: bool) -> None:
         from omegaconf import OmegaConf
 
         vk = get_value_kind(value)
         if vk in (ValueKind.INTERPOLATION, ValueKind.STR_INTERPOLATION):
             return
-
-        if OmegaConf.is_none(value):
-            if key is not None:
-                child = self._get_node(key)
-                if child is not None and not child._is_optional():
-                    self._format_and_raise(
-                        key=key,
-                        value=value,
-                        cause=ValidationError("child '$FULL_KEY' is not Optional"),
-                    )
-            else:
-                if not self._is_optional():
-                    self._format_and_raise(
-                        key=None,
-                        value=value,
-                        cause=ValidationError("field '$FULL_KEY' is not Optional"),
-                    )
-
-        if value == "???":
-            return
-
-        if is_assign and isinstance(value, ValueNode) and self._has_element_type():
-            self._check_assign_value_node(key, value)
+        self._validate_non_optional(key, value)
+        # target must be optional by now. no need to check the type of value if None.
```

Comment:
I find this comment odd, because you are doing that exactly in the following line :)

Reply:
True, I was just moving around the comment with the "linked" code to not lose it. I'll remove it.